dascore/utils/array.py

```
@@ -346,28 +368,151 @@ def _apply_op(array1, array2, operator, reversed=False):
+                except DimensionalityError as er:
+                    msg = f"{operator} failed with units {data_units} and {other_units}"
+                    raise UnitError(msg) from er
+                other_data = in_patch_units.magnitude / data_units.magnitude
+            return _apply_op(patch.data, other_data, operator, reversed), attrs
+        if hasattr(result, "units"):
+            return result.magnitude, attrs.update(data_units=str(result.units))
+        # Result is unitless (e.g., from boolean comparison)
+        return result, attrs.update(data_units=None)
+
+    def _apply_op_units(
+        patch, other, operator, attrs, reversed=False, other_units=None
```

Ruff (0.16.1)

[error] Function argument `reversed` is shadowing a Python builtin

(A002)